src/System.Runtime.InteropServices.PInvoke/src/System/Runtime/InteropServices/PInvokeMarshal.cs

```diff
@@ -0,0 +1,219 @@
+        {
+            Marshal.PtrToStructure<T>(ptr, structure);
+        }
+        public static byte ReadByte(System.IntPtr ptr)
+        {
+            return Marshal.ReadByte(ptr);
+        }
+        public static byte ReadByte(System.IntPtr ptr, int offset)
+        {
+            return Marshal.ReadByte(ptr, offset);
+        }
+        public static byte ReadByte(object ptr, int offset)
+        {
+            return Marshal.ReadByte(ptr, offset);
+        }
+        public static short ReadInt16(System.IntPtr ptr)
+        {
+            return Marshal.ReadInt16(ptr);
+        }
+        public static short ReadInt16(System.IntPtr ptr, int offset)
+        {
+            return Marshal.ReadInt16(ptr, offset);
+        }
+        public static short ReadInt16(object ptr, int offset)
+        {
+            return Marshal.ReadInt16(ptr, offset);
+        }
+        public static int ReadInt32(System.IntPtr ptr)
+        {
+            return Marshal.ReadInt32(ptr);
+        }
+        public static int ReadInt32(System.IntPtr ptr, int offset)
+        {
+            return Marshal.ReadInt32(ptr, offset);
+        }
+        public static int ReadInt32(object ptr, int offset)
+        {
+            return Marshal.ReadInt32(ptr, offset);
+        }
+        public static long ReadInt64(System.IntPtr ptr)
+        {
+            return Marshal.ReadInt64(ptr);
+        }
+        public static long ReadInt64(System.IntPtr ptr, int offset)
+        {
+            return Marshal.ReadInt64(ptr, offset);
+        }
+        public static long ReadInt64(object ptr, int offset)
+        {
+            return Marshal.ReadInt64(ptr, offset);
+        }
+        public static System.IntPtr ReadIntPtr(System.IntPtr ptr)
+        {
+            return Marshal.ReadIntPtr(ptr);
+        }
+        public static System.IntPtr ReadIntPtr(System.IntPtr ptr, int offset)
+        {
+            return Marshal.ReadIntPtr(ptr, offset);
+        }
+        public static System.IntPtr ReadIntPtr(object ptr, int offset)
+        {
+            return Marshal.ReadIntPtr(ptr, offset);
+        }
+        public static System.IntPtr ReallocateMemory(System.IntPtr ptr, int sizeInBytes)
+        {
+            return Marshal.ReAllocCoTaskMem(ptr, sizeInBytes);
+        }
+        public static int SizeOf(object structure)
+        {
+            return Marshal.SizeOf(structure);
+        }
+        public static int SizeOf(System.Type type)
+        {
+            return Marshal.SizeOf(type);
+        }
+        public static int SizeOf<T>()
+        {
+            return Marshal.SizeOf<T>();
+        }
+        public static System.IntPtr StringToBSTR(string s)
+        {
+            return Marshal.StringToBSTR(s);
+        }
+        public static System.IntPtr StringToAllocatedMemoryAnsi(string s)
+        {
+            return Marshal.StringToCoTaskMemAnsi(s);
+        }
+        public static System.IntPtr StringToAllocatedMemoryUni(string s)
```

Comment:
What does "Unicode" mean in this context? Does it mean UTF16? If yes, we should call it StringToAllocatedMemoryUtf16 or something like that.